src/event/mod.rs

```diff
@@ -119,6 +119,24 @@ impl LogEvent {
         util::log::remove(&mut self.fields, &key)
     }
 
+    pub fn remove_prune(&mut self, key: &Atom, drop_empty: bool) -> Option<Value> {
+        let value = util::log::remove(&mut self.fields, &key);
+        if drop_empty && value.is_some() {
+            if let Some(dot) = key.find('.') {
```

Comment:
Is this something we want to support, I know some transforms like kube intentionally use `.` for its keys.

Reply:
This behavior has to be opted into by both the code using `remove_prune` instead of `remove` and then `drop_empty` being turned on.

Reply:
In that case we should either document the gotcha/caveat, or improve our code to support these kinds of keys.

Reply:
I think the fields which are not nested can be inserted directly to the `BTreeMap` (a new  method `insert_flat` can be added to `LogEvent`).

Reply:
Is there a better way to iterate through the path to determine what the parent object is?

Reply:
@bruceg I think it actually might make sense to add `remove_prune` method directly to `LogEvent` (with implementation in `src/event/log/util`), as it seems to be somewhat common functionality. Then it can use, similarly to `remove`, the internal `PathIter` structure to parse the field path notation.